youtube_dl/extractor/bbc.py

```diff
@@ -959,6 +959,32 @@ def _real_extract(self, url):
              r'videoId\s*:\s*["\'](%s)["\']' % self._ID_REGEX],
             webpage, 'vpid', default=None)
 
+        # bbc reel (e.g. https://www.bbc.com/reel/video/p07c6sb6/how-positive-thinking-is-harming-your-happiness)
+        if not programme_id:
+            initial_data = self._search_regex(
+                r'<script[^>]+id="initial-data"[^>]+data-json=\'(.+)\'>',
+                webpage, 'initial data', default=None)
+            if initial_data:
+                # let's see if it actually is, or ExtractorError
+                initial_data = self._parse_json(unescapeHTML(initial_data), 'initial data')
```

Comment:
should not break the extraction if it wasn't able to parse json data.

Reply:
I'm not convinced. At this point we have no pid, because of the test I added. Failing here means the user/developer will be told that the page looked like a Reel page (had a `<script>` with `id` `initial-data` and a `data-json` attribute -- but maybe that's not sufficiently distinctive?) but the content couldn't be extracted. Otherwise they get a more generic 'No extractor for page' error, no?

Reply:
> I'm not convinced. At this point we have no pid, because of the test I added. Failing here means the user/developer will be told that the page looked like a Reel page (had a `<script>` with `id` `initial-data` and a `data-json` attribute -- but maybe that's not sufficiently distinctive?) but the content couldn't be extracted. Otherwise they get a more generic 'No extractor for page' error, no?

no, they would get warning, and then proceed to try other methods that might success.
and you actually should extract more metadata from the `initial_data` not just the pid and then return immidiantly.

Reply:
OK, I've made that a warning and added one for not actually finding a pid in the JSON.

Reply:
no, as i said you would extract the metadata and return immediately.

Reply:
Implemented in  https://github.com/ytdl-org/youtube-dl/pull/28268/commits/db1096f4b92fbf1be6f8393f46b034f8bc09f8bb